### PR TITLE
Update release documentation

### PR DESCRIPTION
**Pull request by anishasthana**

### CODE REVIEW

**Review by DharmitD — COMMENTED** [review]
@HumairAK could you please take a look and confirm that the GHA being taken out isn't required any more? 
Otherwise, changes /lgtm

File: README.md [hunk 1]
```diff
@@ -647,4 +676,100 @@ Refer to this [repo][kubeflow-pipelines-examples] to see examples of different p
 [kubebuilder-docs]: https://book.kubebuilder.io/
 [dspa-yaml]: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/samples/v2/dspa-all-fields/dspa_all_fields.yaml#L77
 [sample-yaml]: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/samples/v2/dspa-simple/dspa_simple.yaml
-[kubeflow-pipelines-examples]: https://github.com/rh-datascience-and-edge-practice/kubeflow-pipelines-examples 
+[kubeflow-pipelines-examples]: https://github.com/rh-datascience-and-edge-practice/kubeflow-pipelines-examples
+
+## How to create a DSP release
```
Comment (HumairAK) [inline README.md:681]: I think this is out of scope off a front page release, can we instead keep the original location, but link to that doc instead to increase visibility?

Comment (anishasthana) [reply README.md]: Done
